torch/csrc/jit/graph_executor.cpp

```diff
@@ -708,6 +711,11 @@ void runNondiffOptimization(std::shared_ptr<Graph>& graph) {
   // Rewrite subgraphs with many MMs into expressions that batch them.
   BatchMM(graph);
 
+  if (at::globalContext().qEngine() == at::kFBGEMM ||
```

Comment:
What happens if the qEngine is changed between when the code is compiled and when it is run? This seems suspicious. The GraphExecutor has no mechanism to invalidate compilation state and shouldn't grow such a mechanism.

Reply:
Where should we put this then?

Reply:
I think we should not be allowed to change qEngine on the fly. I think it's reasonable to be able to retrieve info about target we're running at and specializing for, and context seems to be fitting that well. However, the context should not be mutable in this scheme. @zdevito, does that make sense to you?